pages/uploader.tsx

```diff
@@ -1,150 +1,268 @@
+// import { getFirestore, collection, addDoc } from 'firebase/firestore';
+import { useCallback, useState } from 'react';
+import { FileError, FileRejection, useDropzone } from 'react-dropzone';
+
+import { ref, uploadBytesResumable } from 'firebase/storage';
+import { firebase, storage } from '../firebase/firebase';
+import { getAuth } from 'firebase/auth';
+import {
+  collection,
+  addDoc,
+  getFirestore,
+  setDoc, // eslint-disable-line
+  doc, // eslint-disable-line
+} from 'firebase/firestore';
+
+export interface UploadableFile {
+  file: File;
+  name: string;
+  preview: string;
+  errors: FileError[];
+}
+// TODO: figure out how to type this properly
+// let Url: {
+//   new (url: string | URL, base?: string | URL | undefined): URL;
+//   createObjectURL: any;
+//   prototype?: URL;
+//   revokeObjectURL?: (url: string) => void;
+// };
+let Url: any;
+if (typeof window !== 'undefined') {
+  Url = window.URL || window.webkitURL;
+}
+const Uploader: NextPage = () => {
+  getAuth();
+
+  const db = getFirestore(firebase);
+  const [file, setFile] = useState<UploadableFile>();
+  const [uploadProgress, setUploadProgress] = useState<string>();
+
+  const [title, setTitle] = useState<string>('');
+  const [subtitle, setSubtitle] = useState<string>('');
+  const [date, setDate] = useState<string>('');
+  const [description, setDescription] = useState<string>('');
+  const [speaker, setSpeaker] = useState<string>('');
+  const [scripture, setScripture] = useState<string>('');
+  const [topic, setTopic] = useState<string>('');
+
+  const onDrop = useCallback(
+    (acceptedFiles: File[], rejectedFiles: FileRejection[]) => {
+      // Do something with the files
+      if (rejectedFiles.length > 0) {
+        // console.log(rejectedFiles[0].errors[0]);
+        setFile(undefined);
+        return;
+      }
+      // const reader = new FileReader();
+      // reader.readAsDataURL(acceptedFiles[0]);
+      // reader.addEventListener('progress', function (pe) {
+      //   if (pe.lengthComputable) {
+      //     console.log('Progress:', pe.loaded, 'Total:', pe.total);
+      //   }
+      // });
+      // reader.addEventListener(
+      //   'load',
+      //   function () {
+      const mappedAccepted = {
+        file: acceptedFiles[0],
+        // preview: reader.result as string,
+        preview: Url.createObjectURL(acceptedFiles[0]),
+        name: acceptedFiles[0].name.replace(/\.[^/.]+$/, ''),
+        errors: [],
+      };
+      setFile(mappedAccepted);
+      //   },
+      //   false
+      // );
+      // const mappedAccepted2 = acceptedFiles.map((file) => ({
+      //   file,
+      //   errors: [],
+      // }));
+      // setFile((curr) => [...curr, mappedAccepted, ...rejectedFiles]);
+    },
+    []
+  );
+
+  const { getRootProps, getInputProps } = useDropzone({
+    onDrop,
+    multiple: false,
+    accept: ['audio/*'],
+  });
+
+  function uploadFile(file: File) {
+    const sermonRef = ref(storage, `sermons/${file.name}`);
```

Comment:
We don't want to use the file name here as it could be anything. Probably want to use the title or a randomly generated uuid so that we can allow for multiple sermons to have the same title say in the case that two different priests have a sermon with the same name.